runtime/dancebox/src/lib.rs

```diff
@@ -309,8 +309,38 @@ impl pallet_timestamp::Config for Runtime {
     type WeightInfo = ();
 }
 
+pub struct FindAuthorFromDigest;
+
+impl FindAuthor<u32> for FindAuthorFromDigest {
+    fn find_author<'a, I>(digests: I) -> Option<u32>
+    where
+        I: 'a + IntoIterator<Item = (sp_runtime::ConsensusEngineId, &'a [u8])>,
+    {
+        use {
+            parity_scale_codec::Decode,
+            sp_consensus_aura::{Slot, AURA_ENGINE_ID},
+        };
+
+        for (id, mut data) in digests.into_iter() {
+            if id == AURA_ENGINE_ID {
+                let slot = Slot::decode(&mut data).ok()?;
+
+                let authorities =
+                    AuthorityAssignment::collator_container_chain(Session::current_index())
+                        .expect("authorities should be set")
+                        .orchestrator_chain;
+
+                let author_index = *slot % authorities.len() as u64;
+                return Some(author_index as u32);
+            }
+        }
+
+        None
+    }
+}
+
 impl pallet_authorship::Config for Runtime {
```

Comment:
pallet-authorship is removed from construct-runtime, so we can probably remove it too and forget about this

Reply:
It's unfortunate that we don't get a "unused code" warning